src/classes/ReminderProcessor.cls

```diff
@@ -0,0 +1,27 @@
+global class ReminderProcessor  implements Database.Batchable<sObject>, Database.Stateful{
+        
+    global Database.QueryLocator start (Database.BatchableContext bc){
+        return Database.getQueryLocator ('SELECT Id, Draft__c, Due_date__c, Email_Sent__c, Account__r.Email__c, Seller__r.Company_Name__c FROM Invoice__c WHERE Status__c = \'Unpaid\'');
+    }
+    
+    global void execute (Database.BatchableContext bc, List<Invoice__c> scope){
+	List<Notify_Purchaser__c> nfList = new List <Notify_Purchaser__c>();		      
+        if(scope!=null || !scope.isEmpty()){
```

Comment:
To sprawdzanie zbędne, jak nie pobierze rekordów to metoda execute się nie uruchomi.

Reply:
tylko w tym wypadku czy jeszcze sa inne metody dla ktorych nie musze stosowac takich zabezpieczen?

Reply:
Tylko tutaj, jak mi się jakaś przypomni to przegadamy sobie w poniedziałek.